contracts/secret-storage/SecretManager.sol

```diff
@@ -364,50 +365,120 @@ contract SecretManager is
     ) internal {
         UserStorage memory userStoreData = userStorage[_secretId];
         bool ackFailed;
+        bool reselectedAckFailed;
+        bool hasReplacedStores = _checkIfSecretHasReplacedStores(_secretId);
+
         uint256 len = userStoreData.selectedEnclaves.length;
         for (uint256 index = 0; index < len; index++) {
+            address enclaveAddress = userStoreData.selectedEnclaves[index].enclaveAddress;
             if(block.timestamp <= userStoreData.selectedEnclaves[index].selectTimestamp + ACKNOWLEDGEMENT_TIMEOUT)
```

Comment:
check the case where 2 enclaves are dead. they can have different selection time. Currently we won't be able to call ack-failed for the earlier dead enclave.